staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -150,6 +150,24 @@ func encodeManagedFields(managedFields fieldpath.ManagedFields) (encodedManagedF
 }
 
 func sortEncodedManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (sortedManagedFields []metav1.ManagedFieldsEntry, err error) {
+	operation := func(p, q metav1.ManagedFieldsEntry) bool {
```

Comment:
I think you're going too far with the abstractions here. You could move this logic directly in the managedFieldsSorter (I don't think we'll want multiple of these), or that we want to configure the sorting algorithm dynamically.